src/playlist-loader.js

```diff
@@ -96,7 +96,12 @@
 
       // a flag that disables "expired time"-tracking this setting has
       // no effect when not playing a live stream
-      this.trackExpiredTime_ = false;
+      loader.trackExpiredTime_ = false;
+
+      // a flag that signals if we are in the process of retrying the
+      // process of fetching a live playlist because the last fetch
+      // introduced no changes
+      loader.retryUpdate_ = false;
```

Comment:
Damnit you noticed! I couldn't think of a better way to detect that we have failed (twice) to detect changes.